include/deal.II/fe/fe_interface_values.h

```diff
@@ -0,0 +1,686 @@
+
+/**
+ * FEInterfaceValues is a data structure to access and assemble Finite Element
+ * data on interfaces between two cells of a mesh.
+ *
+ * It provides a way to access averages, jump terms, and similar operations used
+ * in Discontinuous Galerkin methods on a face between two neighboring cells.
+ * This allows the computation of typical mesh-dependent linear or bilinear
+ * forms in a similar way as FEValues does for cells and FEFaceValues does for
+ * faces. In
+ * the literature, the faces between neighboring cells are called "inner
+ * interfaces" or "facets".
+ *
+ * Internally, this class provides an abstraction for two FEFaceValues
+ * objects (or FESubfaceValues when using adaptive refinement). The class
+ * introduces a new "interface dof index" that walks over
+ * the union of the dof indices of the two FEFaceValues objects. Helper
+ * functions allow translating between the new "interface dof index" and the
+ * corresponding "cell index" (0 for the first cell, 1 for the second cell)
+ * and "dof index" within that cell.
+ *
+ * The class is made to be used inside MeshWorker::mesh_loop. It is intended
+ * to be a low level replacement for MeshWorker and LocalIntegrators and a
+ * higher level abstraction compared to assembling face terms manually.
+ *
+ * @author Timo Heister, 2019.
+ */
+template <int dim, int spacedim = dim>
+class FEInterfaceValues
+{
+public:
+  /**
+   * Construct an FEInterfaceValues from existing FEFaceValues objects.
+   *
+   * The arguments to this functions are used to construct the internal
+   * FEFaceValues and FESubfaceValues objects for the two cells of the
+   * interface. As the arguments are copied, there are no requirements
+   * on the lifetime of the arguments.
+   */
+  FEInterfaceValues(const FEFaceValues<dim> &   fe_face_values,
+                    const FESubfaceValues<dim> &fe_subface_values);
+
+  /**
+   * Construct the FEInterfaceValues with a single FiniteElement (same on both
+   * sides of the facet). The FEFaceValues objects will be initialized with
+   * the given @p mapping, @p quadrature, and @p update_flags.
+   */
+  FEInterfaceValues(const Mapping<dim, spacedim> &      mapping,
+                    const FiniteElement<dim, spacedim> &fe,
+                    const Quadrature<dim - 1> &         quadrature,
+                    const UpdateFlags                   update_flags);
+
+  /**
+   * Construct the FEInterfaceValues with a single FiniteElement and
+   * a Q1 Mapping.
+   *
+   * See the constructor above.
+   */
+  FEInterfaceValues(const FiniteElement<dim, spacedim> &fe,
+                    const Quadrature<dim - 1> &         quadrature,
+                    const UpdateFlags                   update_flags);
+
+  /**
+   * Re-initialize this object to be used on a new interface given by two faces
+   * of two neighboring cells.
+   *
+   * The arguments (including their order) is identical to the @p face_worker arguments
+   * in MeshWorker::mesh_loop.
```

Comment:
Just remove this sentence -- the function is easy enough to understand without having to look up something else. But if you want to keep it:
```suggestion
   * in MeshWorker::mesh_loop().
```
and replace is->are in the line above.